content/ui-ux.md

```diff
@@ -0,0 +1,514 @@
+5. How to build components that can cope with a variety of different data sources
+6. How to use animation to keep users informed of changes
+
+## Reusable components
+
+Regardless of the rendering library that you are using, there are some patterns in how you build your User Interface (UI) that will help make your app's code easier to understand, test, and maintain. These patterns, much like general patterns of modularity, revolve around making the interfaces of your UI elements very clear, and avoiding using techniques that bypass these known interfaces.
+
+In this article, we'll refer to the elements in your user interface as "components". Although in some systems, you may refer to them as "templates", it can be a good idea to think of them as something more modular like a component which has an API, rather than a template which is usually seen in a looser way.
+
+To begin with, let's consider two categories of components that are useful to think about:
+
+### Pure Components
+
+A "pure" component is a component which doesn't rely on anything apart from it's inputs (it's *template arguments* in Blaze, or *props* in React) to render. 
+
+In Meteor, specifically this means a component which does not access data from any global sources (typically either Collections or Stores). For instance, in the Todos example app, the `todosItem` template takes in the todo that it is rendering and does not ever look directly in the `Todos` collection.
+
+The advantages of pure components are the following:
+
+ 1. They are easy to reason about---you don't need to understand how the data in the global store changes, simply how the arguments to the component change.
+
+ 2. They are easy to test---you don't need to be careful about the environment you render them in, all you need to do is provide the right arguments.
+
+ 3. They are easy to styleguide---as we'll see in the next section, when styleguiding components, a clean environment makes things much easier to work with.
+
+ 4. They are a lot more flexible, and thus can be re-used without requiring re-working.
+
+ ### Global Data stores
+
+ So which are the global data stores that you should be avoiding in pure components? There are a few. Meteor as a framework is built with ease of development in mind, which typically means you can access a lot of things globally. Although this is very useful when building "smart" components (see below), it's a good idea to avoid it in pure ones:
+
+  - Your collections, as well as the `Meteor.users` collection,
+  - Accounts information, like `Meteor.user()` and `Meteor.loggingIn()`
+  - Current route information
+  - Any other client-side data stores (see XXX not sure which article? maybe this one)
+
+### Smart Components
+
+Of course sometimes you do need to access the global data sources to feed data into your application. We call components that need to access data "smart". Such components typically the following things
```

Comment:
> Of course sometimes you do need to access the global data sources to feed data into your application.

"While most of the components in your app should be pure and reusable, they need to get their data passed in from somewhere. This is where "smart" components come in."